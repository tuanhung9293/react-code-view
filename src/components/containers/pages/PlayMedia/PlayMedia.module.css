.dismiss {
    text-align: right;
}

.img {
    max-width: 95%;
    left: 50%;
    position: absolute;
    transform: translateX(-50%);
}

.audio {
    width: 80%;
    left: 50%;
    top: 50%;
    position: absolute;
    transform: translate(-50%, -50%);
}