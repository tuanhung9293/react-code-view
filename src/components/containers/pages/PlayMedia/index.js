import React from 'react'
import DocumentTitle from 'components/html/DocumentTitle'
import {connect} from 'react-redux'
import {withRouter} from 'react-router-dom'
import DismissButton from 'components/ui/buttons/DismissButton'
import {Player} from 'video-react'
import ReactAudioPlayer from 'react-audio-player'
import styles from './PlayMedia.module.css'

class PlayMedia extends React.Component {
  render() {
    const {mediaSource, windowSize, history} = this.props
    return (
      <div>
        <div className={styles.dismiss}>
          <DismissButton onClick={history.goBack} />
        </div>
        <DocumentTitle title="Play Media" />
        {mediaSource.mediaType === 'video' &&
          <Player
            playsInline
            autoPlay
            src={mediaSource.mediaUrl}
          />
        }
        {mediaSource.mediaType === 'image' &&
          <img className={styles.img}
            src={mediaSource.mediaUrl}
            style={{maxHeight: windowSize.height * 0.9}}
          />
        }
        {mediaSource.mediaType === 'audio' &&
          <ReactAudioPlayer
            src={mediaSource.mediaUrl}
            autoPlay
            controls
            className={styles.audio}
          />
        }
      </div>
    )
  }
}

function mapStateToProps(state, {match}) {
  return {
    mediaSource: state[match.params.source].data.find(item => item.nasaId === match.params.mediaId) || {},
    windowSize: state.ui.windowSize
  }
}

export default withRouter(connect(mapStateToProps, null)(PlayMedia))
