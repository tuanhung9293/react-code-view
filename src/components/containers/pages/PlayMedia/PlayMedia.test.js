import React from 'react'
import {shallow} from 'enzyme'
import PlayMedia from './'

describe('renders', () => {
  it('renders PlayMedia', () => {
    expect(shallow(
      <PlayMedia />
    )).toMatchSnapshot()
  })
})

