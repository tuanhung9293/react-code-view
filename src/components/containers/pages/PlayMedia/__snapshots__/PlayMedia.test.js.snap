// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders renders PlayMedia 1`] = `
<Route>
  <Component />
</Route>
`;
